includes/class-wc-payments-utils.php

```diff
@@ -839,15 +831,21 @@ public static function format_explicit_currency(
 	): string {
 		$currency = strtoupper( $currency );
 
-		$formatted_amount = wc_price(
-			$amount,
-			wp_parse_args( $currency_format, self::get_currency_format_for_wc_price( $currency ) )
-		);
-
+		$currency_format  = wp_parse_args( $currency_format, self::get_currency_format_for_wc_price( $currency ) );
+		$formatted_amount = wc_price( $amount, $currency_format );
 		$formatted_amount = html_entity_decode( wp_strip_all_tags( $formatted_amount ) );
 
 		if ( $skip_symbol ) {
-			$formatted_amount = preg_replace( '/[^0-9,\.]+/', '', $formatted_amount );
+			if ( isset( $currency_format['thousand_separator'], $currency_format['decimal_separator'] ) ) {
```

Comment:
This is an extra fix for correctly keeping the decimal and thousand separators while replacing all other things.